File: java-back-server/src/main/java/space/borisgk/taxi/api/model/dto/MessageDto.java
package space.borisgk.taxi.api.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import space.borisgk.taxi.api.model.MessageStatus;
import space.borisgk.taxi.api.model.MessageType;
import space.borisgk.taxi.api.model.entity.Trip;
import space.borisgk.taxi.api.model.entity.User;

import javax.persistence.*;
import java.util.Date;

@Builder
@Setter
@Getter
@AllArgsConstructor
public class MessageDto {
    private String id;

    private String userId;
    private String tripId;
    /**
     * base64 data
     */
    private String data;
    private Date date;
    private MessageStatus status;
    private MessageType type;
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/model/dto/MessageDto.kt
package space.borisgk.taxi.api.model.dto

import lombok.Builder
import lombok.Getter
import lombok.Setter
import space.borisgk.taxi.api.model.entity.Trip
import space.borisgk.taxi.api.model.entity.User

import javax.persistence.*
import java.util.Date

class MessageDto {
    var id: Int? = null
        set(id) {
            field = this.id
        }

    var userId: Int? = null
        set(userId) {
            field = this.userId
        }
    var tripId: Int? = null
        set(tripId) {
            field = this.tripId
        }
    var message: String? = null
        set(message) {
            field = this.message
        }
    var date: Date? = null
        set(date) {
            field = this.date
        }
}
